lib/internal/perf/observe.js

```diff
@@ -380,55 +371,45 @@ function clearEntriesFromBuffer(type, name) {
   } else {
     return;
   }
+
   if (name === undefined) {
-    resetBuffer(buffer);
+    resetBuffer(type);
     return;
   }
 
-  let head = null;
-  let tail = null;
-  let count = 0;
-  for (let entry = buffer.head; entry !== null; entry = entry[kBufferNext]) {
-    if (entry.name !== name) {
-      head = head ?? entry;
-      tail = entry;
-      continue;
-    }
-    if (tail === null) {
-      continue;
+  for (let i = 0; i < buffer.length;) {
```

Comment:
can't this be reassigning + `filter` instead of `splice`? It allocates but I suspect it may still be faster?